crates/net/network/src/config.rs

```diff
@@ -73,7 +74,7 @@ pub struct NetworkConfig<C, N: NetworkPrimitives = EthNetworkPrimitives> {
     /// The executor to use for spawning tasks.
     pub executor: Box<dyn TaskSpawner>,
     /// The `Status` message to send to peers at the beginning.
-    pub status: Status,
+    pub status: StatusMessage,
```

Comment:
this is a bit problematic because we can't convert this in both directions, for example if we configure 69, we can really convert this fully to <=68

so I think we need to keep Status here and in addition to that we need to a way to configure the block range for 69?
which we eventually need to set externally if there's history expiry.

so we can introduce two more fields here for the block range?

Reply:
I feel you on the loss of fidelity if try to bi-directionally convert. I updated this to use a `UnifiedStatus`, which I'll elaborate on a couple comments down

Reply:
and for being able to externally set the block range, we should be able to use `UnifiedStatus::set_history_range`